### PR TITLE
docs: Revert using cluster-internal for getting started guide

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Reverting parts of https://github.com/stackabletech/hdfs-operator/pull/492

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: docs/modules/hdfs/examples/getting_started/hdfs.yaml [hunk 1]
```diff
@@ -11,13 +11,13 @@ spec:
     dfsReplication: 1
   nameNodes:
     config:
-      listenerClass: cluster-internal
+      listenerClass: external-stable
```
Comment (razvan) [inline docs/modules/hdfs/examples/getting_started/hdfs.yaml]: why is this needed?

Comment (sbernauer) [reply docs/modules/hdfs/examples/getting_started/hdfs.yaml]: It is not strictly needed, but I would argue users want to able to access the stuff the just deployed ;)
And port-forwards IMHO should be a thing of last resort as they break all the time.

In this concrete HDFS case users accessing the WebUIs, for Trino and Superset it would be submitting queries or creating Dashboards,

Comment (razvan) [reply docs/modules/hdfs/examples/getting_started/hdfs.yaml]: I was confused when debugging  because this particular guide doesn't access HDFS from outside the cluster. Also the doc doesn't explain what the `listenerClass` even does (which is fine in this context) , that's why I changed it to the "safer" value.

Comment (sbernauer) [reply docs/modules/hdfs/examples/getting_started/hdfs.yaml]: I see. The default is actually `cluster-internal` for security reasons, so we need to set it explicitly in case we want external access.

Would you be in favor of internal or external access here?

Comment (razvan) [reply docs/modules/hdfs/examples/getting_started/hdfs.yaml]: > Would you be in favor of internal or external access here?

In the current state of the guide I would prefer "internal". If the guide starts explaining the listener class and/or adds instructions on how to access the cluster from outside, then yes, change it to "external".

Comment (sbernauer) [reply docs/modules/hdfs/examples/getting_started/hdfs.yaml]: I think we need some sort of strategy/guideline what we want to do, created https://github.com/stackabletech/decisions/issues/7. Happy about comments!

**Review by razvan — APPROVED** [review]
as discussed. thanks.

**Comment by sbernauer** [comment]
Thanks!
